autorest/codegen/__init__.py

```diff
@@ -135,12 +139,37 @@ def _build_package_dependency() -> Dict[str, str]:
             "dependency_msrest": "msrest>=0.6.21",
         }
 
+    @staticmethod
+    def _build_security(yaml_data: Dict[str, Any]) -> Dict[str, Any]:
+        security_yaml = yaml_data.get("security")
+        security: Dict[str, Any] = dict()
+        if security_yaml and security_yaml.get("authenticationRequired"):
+            # the key word is "AADToken" and "AzureKey"
+            aad_token = set()
+            azure_key = ""
+            for scheme in security_yaml.get("schemes"):
+                if _AAD_TOKEN == scheme["type"]:
+                    aad_token.update(scheme["scopes"])
+                elif _AZURE_KEY == scheme["type"]:
+                    # only accept the last one
+                    azure_key = scheme["headerName"]
+            if aad_token:
+                security[_AAD_TOKEN] = list(aad_token)
+            if azure_key:
+                security[_AZURE_KEY] = azure_key
+        return security
+
     def _create_code_model(self, yaml_data: Dict[str, Any], options: Dict[str, Union[str, bool]]) -> CodeModel:
         # Create a code model
 
         code_model = CodeModel(options=options)
+        security = self._build_security(yaml_data)
```

Comment:
I think the way you've structured the `if elif` can lead to some cases not being handled correctly.

For example, if I include the credential scopes using the swagger, but I also use `--add-credential` (which defaults to AAD auth). Then, we don't get the scopes that the user added to the swagger because we also specified a flag.

I feel like it's easier to just do security first, update as many of the credential properties on `code_model` as you can, then look for flags / azure-arm and update those properties as well

Reply:
as https://github.com/Azure/autorest.python/pull/1161#discussion_r823290951, we shall avoid handling credential flags together with security definition. Now that modelerfour has supported  definition in swagger, we shall encourage to use security definition. It will confusing for customers to use both of them together and complicated for us to handle them together.

Reply:
what I'm trying to say is we can't prevent users from using them together, so to make our code cleaner and still work, so what I'm saying is it would be great to structure the code like this:
1. Have a shared function that takes in credential policies (regardless of where they come from), and set the credential policy of code model based on that
2. get the credential polic(ies) from security, pass to shared function
3. get the credential polic(ies) from old flags, pass to shared function

it would require restructuring some existing code too, but what we really want to do is duplicate as little code as possible. Additionally, if you think about "what is the difference between supplying credentials through security or through the legacy flags", there should be no difference in behavior whatsoever, just in how we get the information. Therfore, it's really great to structure the code like that, where the only differences are "how do we get the credential policies that users inputted"

Reply:
this way also when we add support for multiple authentication policies, all the code you would need to change would be in the one shared function between the two

Reply:
The solution will actually approve users to mix credential flags and security definition. While, it is not big deal. Let me put the debate aside to refactor current code.

Reply:
@iscai-msft I have updated the code including the refactor, please review.